lib/data/multipleBackendGateway.js

```diff
@@ -215,9 +215,9 @@ const multipleBackendGateway = {
     },
 
     objectTagging: (method, key, bucket, objectMD, log, cb) => {
-        const location = objectMD.dataStoreName;
+        const location = objectMD.dataStoreName || undefined;
```

Comment:
the `|| undefined` is for backwards compatibility. I'll add a comment. And we use the `dataStoreName` attribute from the putObj and deleteObj APIs. A better fix would be to get rid of the `objectMD.controllingLocationConstraint` param, but it's used for other MPU things, and for the sake of expediency of the fix, I opted for having both.